SecretInAFile: strip newlines by default

Lekensteyn opened the pull request:
The secrets provider can be used to read passwords which likely do not
have trailing newlines. Strip them by default, this makes it easier
when editing via vim or using a command like `echo sikrit > foo`.

Add a strip option just in case you want to override this behavior. Note
that for some reason "\r" are converted to newlines anyway, so perhaps
this option can just be removed completely.

Fixes https://github.com/buildbot/buildbot/issues/3178

## Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [x] I have updated the appropriate documentation

tardyp commented on master/buildbot/secrets/providers/file.py:
```diff
@@ -58,22 +58,25 @@ def loadSecrets(self, dirname, suffixes):
                     if suffix:
                         secretfile = secretfile[:-len(suffix)]
+                    if strip:
+                        secretvalue = secretvalue.rstrip("\r\n")
```
I would strip all the whitespaces the python way . secretvalue.strip()

Lekensteyn replied:
That would not work with passwords that start with a space (yes, I have at least one such password).

tardyp commented on master/docs/developer/secrets.rst:
```diff
@@ -52,7 +52,7 @@ File provider
 
 In the master configuration the provider is instantiated through a Buildbot service secret manager with the file directory path.
-File secrets provider reads the file named by the key wanted by Buildbot and returns the contained text value.
+File secrets provider reads the file named by the key wanted by Buildbot and returns the contained text value (removing trailing newlines if present).
```
removing leading and trailing whitespace

tardyp commented on master/docs/manual/secretsmanagement.rst:
```diff
@@ -63,7 +63,8 @@ SecretInAFile
 
 In the passed directory, every file contains a secret identified by the filename.
 
```
we need to document the full signature here

tardyp commented:
@Lekensteyn Could you please update the documentation as suggested?

Lekensteyn commented:
@tardyp Done, is this good enough or do you want the `strip=True` parameter to be included too oin the function call?
```diff
diff --git a/master/docs/manual/secretsmanagement.rst b/master/docs/manual/secretsmanagement.rst
index a6876d74a..344b23070 100644
--- a/master/docs/manual/secretsmanagement.rst
+++ b/master/docs/manual/secretsmanagement.rst
@@ -64,7 +64,7 @@ SecretInAFile
 In the passed directory, every file contains a secret identified by the filename.
 
 e.g: a file ``user`` contains the text ``pa$$w0rd``. Trailing newlines are
-removed from the contents.
+removed from the contents (unless ``strip=False``).
 
 SecretInVault
 `````````````
```

tardyp commented:
@Lekensteyn I would like the documentation to document the actual signature for SecretsInAFile

Lekensteyn commented:
@tardyp Done (also fixed a syntax error in the existing example)

tardyp commented on master/docs/manual/secretsmanagement.rst:
```diff
@@ -59,11 +59,13 @@ SecretInAFile
 In the passed directory, every file contains a secret identified by the filename.
 
-e.g: a file ``user`` contains the text ``pa$$w0rd``.
+e.g: a file ``user`` contains the text ``pa$$w0rd``. Trailing newlines are
```
please look at other part of the doc for example on how we describe the signature of a step ex:
```

.. py:class:: buildbot.steps.source.git.Git

The :bb:step:`Git` build step clones or updates a `Git <http://git.or.cz/>`_ repository and checks out the specified branch or revision.


.. note::

    The Buildbot supports Git version 1.2.0 and later: earlier versions (such as the one shipped in Ubuntu 'Dapper') do not support the :command:`git init` command that the Buildbot uses.

::

    from buildbot.plugins import steps

    factory.addStep(steps.Git(repourl='git://path/to/repo', mode='full',
                              method='clobber', submodules=True))

The Git step takes the following arguments:

``repourl``
   (required): the URL of the upstream Git repository.

``branch``
   (optional): this specifies the name of the branch to use when a Build does not provide one of its own.
   If this parameter is not specified, and the Build does not provide a branch, the default branch of the remote repository will be used.

[...]
```

Lekensteyn commented:
Thanks @tardyp , that was a good pointer, I've added a list of arguments.
